File: backend/go/internal/models/address.go
package models

type Address struct {
	ID          uint    `gorm:"primaryKey;column:id" json:"id"`
	ContactID   uint    `gorm:"column:contact_id" json:"contactId"`
	Street      string  `gorm:"column:street" json:"street"`
	City        *string `gorm:"column:city" json:"city"`
	District    *string `gorm:"column:district" json:"district"`
	SubDistrict *string `gorm:"column:sub_district" json:"subDistrict"`
	PostalCode  *string `gorm:"column:postal_code" json:"postalCode"`
	CreatedAt   int64   `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt   int64   `gorm:"column:updated_at" json:"updatedAt"`

	Contact Contact `gorm:"foreignKey:ContactID" json:"contact,omitempty"`
}

File: backend/ASPNET/ASPNET/Models/Entities/Address.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ASPNET.Models.Entities
{
    public class Address
    {
        public int Id { get; set; }
        
        [Required]
        public int ContactId { get; set; }
        
        [Required]
        [StringLength(255)]
        public string Street { get; set; } = string.Empty;
        
        [StringLength(100)]
        public string? City { get; set; }
        
        [StringLength(100)]
        public string? District { get; set; }
        
        [StringLength(100)]
        public string? SubDistrict { get; set; }
        
        [StringLength(10)]
        public string? PostalCode { get; set; }
        
        [Required]
        public long CreatedAt { get; set; }
        
        [Required]
        public long UpdatedAt { get; set; }

        // Navigation properties
        [ForeignKey("ContactId")]
        public virtual Contact Contact { get; set; } = null!;
    }
}